pkg/docker/docker.go

```diff
@@ -195,13 +196,7 @@ func (c *Client) GetOutputStream(ctx context.Context, id string, since string, f
 
 func (c *Client) RemoveContainer(ctx context.Context, id string) error {
 	log.Ctx(ctx).Debug().Str("id", id).Msgf("Container Stop")
-	timeout := time.Millisecond * 100
-	if err := c.ContainerStop(ctx, id, timeout); err != nil {
-		if dockerclient.IsErrNotFound(err) {
-			return nil
-		}
-		return errors.WithStack(err)
-	}
+	// ContainerRemove kills and removes a container from the docker host.
```

Comment:
Is it safe to do this without calling stop first to give it a chance to shut down gracefully?  
From what I can tell Stop sends a SIGINT (or SIGTERM),  but kill will send SIGKILL (which is uncatchable).
Maybe in a future PR we should look at the docker lifecycle.

Reply:
I removed the above lines as I didn't like the arbitrary timeout for stopping the container, I expected this method wrapper to _only_ remove the container and was surprised to see it doing more. If we want the extra stop call back, I can add it.

The docker `executionHander` uses its `kill` method to stop the container, and internally `kill` calls `ContainerStop`. We only use `RemoveContainer` in `destroy`, which calls `kill` before removing the container and associated objects. I have left a [TODO](https://github.com/bacalhau-project/bacalhau/blob/frrist/async-executor/pkg/executor/docker/handler.go#L151) in the aforementioned `kill` method regarding the need for better lifecycle management. In future changes we should allow a signal to be sent to the container when stopping it. The underlying docker client supports this behavior, we just need to wire it up.